tests/foreman/cli/test_registration.py

```diff
@@ -224,3 +226,42 @@ def test_negative_global_registration_without_ak(module_target_sat):
         'Failed to generate registration command:\n  Missing activation key!'
         in context.value.message
     )
+
+
+def test_positive_custom_facts_for_host_registration(
+    module_sca_manifest_org,
+    module_location,
+    module_target_sat,
+    rhel8_contenthost,
+    module_activation_key,
+):
+    """Attempt to register a host and check all the interfaces are created from the custom facts
+
+    :id: db73c146-4557-4bf4-a8e2-950ecba31620
+
+    :steps:
+        1. Register the host.
+        2. Check the host is registered and all the interfaces are created successfully.
+
+    :expectedresults: Host registered successfully with all interfaces created from the custom facts.
+
+    :BZ: 2250397
+
+    :customerscenario: true
+    """
+    facts = {
+        f"net.interface.{gen_string('alphanumeric')}.mac_address": gen_mac(),
+        f"net.interface.{gen_string('alpha')}.{gen_string('numeric')}.mac_address": gen_mac(),
+        f"net.interface.{gen_string('alpha')}.{gen_string('numeric')}.mac_address": gen_mac(),
```

Comment:
Could you use different combinations of this interface names as mentioned in BZ instead of gen_string?

Reply:
I can but as per the BZ, any one uppercase letter in the interface would work and gen_string exactly does that.

Reply:
I think `gen_string('alpha')` is not guaranteed to contain an uppercase letter:
https://fauxfactory.readthedocs.io/en/latest/api.html#fauxfactory.factories.strings.gen_alpha